test/w3_header_utils_test.dart

```diff
@@ -0,0 +1,62 @@
+import 'dart:math';
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:instabug_flutter/src/models/generated_w3c_header.dart';
+import 'package:instabug_flutter/src/utils/w3c_header_utils.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'w3_header_utils_test.mocks.dart';
+
+@GenerateMocks([Random])
+void main() {
+  final mRandom = MockRandom();
+
+  setUpAll(() {
+    W3CHeaderUtils.$setRandom(mRandom);
+  });
+  setUp(() {
+    when(mRandom.nextInt(any)).thenReturn(217222);
+  });
+
+  tearDown(() {
+    reset(mRandom);
+  });
+
+  test('generateTracePartialId should generate a non-zero hex string', () {
+    final hexString = W3CHeaderUtils.generateTracePartialId().hexPartialId;
+    expect(hexString, isNot('00000000'));
```

Comment:
This test will always pass even if we don't add checks for non-zero partial ID in the code because the mocked `Random` instance always returns a non-zero value.

Instead, let's mock it to return `0` on the first call and a non-zero value on the second call for example to make sure it won't accept the zero value.